### PR TITLE
System.AppContext F# snippets

### PR DESCRIPTION
**Pull request by albert-du**
## Summary

Adds F# snippets for System.AppContext

### CODE REVIEW

**Comment by albert-du** [comment]
@dsyme could you review this?

File: samples/snippets/fsharp/VS_Snippets_CLR_System/System.AppContext.Class/fs/Example6.fs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        fullString.IndexOf(substr, StringComparison.CurrentCulture)
+
+// </Snippet6>
+
+// <Snippet7>
+let value = "The archaeologist"
+let substring = "archæ"
+
+let position =
+    StringLibrary.substringStartsAt value substring
```
Comment (dsyme) [inline samples/snippets/fsharp/VS_Snippets_CLR_System/System.AppContext.Class/fs/Example6.fs:21]: Why define "StringLibrary"?  Just call "IndexOf directly?

**Review by dsyme — APPROVED** [review]

File: samples/snippets/fsharp/VS_Snippets_CLR_System/System.AppContext.Class/fs/Example6.fs (hunk 1 above)
Comment (albert-du) [reply samples/snippets/fsharp/VS_Snippets_CLR_System/System.AppContext.Class/fs/Example6.fs:21]: The article demonstrates how a library may use Appcontext switches to vary calling behavior. Snippets 4 and 6 demonstrate different iterations of the same library with different behavior. Snippet 8 shows how to choose behavior based on a switch. While the calling IndexOf directly would work, it wouldn't fit with the article.

**Review by BillWagner — APPROVED** [review]
Thanks again @albert-du 

I'll :shipit: now.

**Pull request merged**